src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -105,10 +113,37 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
         Course updatedCourse = editPersonDescriptor.getCourse().orElse(personToEdit.getCourse());
         Tag updatedTag = editPersonDescriptor.getTag().orElse(personToEdit.getTag());
+
         ArrayList<Module> updatedModules = editPersonDescriptor.getModules().orElse(personToEdit.getModules());
+        if (editPersonDescriptor.hasModuleChanges()) {
```

Comment:
Thoughtful implementation of `createEditPerson` method by checking whether the module has a grade added to it.